include/boost/math/quadrature/gauss.hpp

```diff
@@ -29,14 +29,37 @@ struct gauss_constant_category
       (std::numeric_limits<T>::is_specialized == 0) ? 999 :
       (std::numeric_limits<T>::radix == 2) ?
       (
-         (std::numeric_limits<T>::digits <= std::numeric_limits<float>::digits) && std::is_convertible<float, T>::value ? 0 :
-         (std::numeric_limits<T>::digits <= std::numeric_limits<double>::digits) && std::is_convertible<double, T>::value ? 1 :
-         (std::numeric_limits<T>::digits <= std::numeric_limits<long double>::digits) && std::is_convertible<long double, T>::value ? 2 :
 #ifdef BOOST_HAS_FLOAT128
-         (std::numeric_limits<T>::digits <= 113) && std::is_constructible<__float128, T>::value ? 3 :
+         (std::numeric_limits<T>::digits <= 113) && std::is_constructible<__float128, T>::value ? 0 :
+#else
+         (std::numeric_limits<T>::digits <= std::numeric_limits<long double>::digits) && std::is_convertible<long double, T>::value ? 0 :
 #endif
          (std::numeric_limits<T>::digits10 <= 110) ? 4 : 999
       ) : (std::numeric_limits<T>::digits10 <= 110) ? 4 : 999;
+   
+   using storage_type =
+      typename std::conditional<(std::numeric_limits<T>::is_specialized == 0), T,
+         typename std::conditional<(std::numeric_limits<T>::radix == 2),
+            typename std::conditional< ((std::numeric_limits<T>::digits <= std::numeric_limits<float>::digits) && std::is_convertible<float, T>::value),
```

Comment:
The math library requires a c++14 compiler now so you should be able to use `std::conditional_t` now to remove the typename and ::type

Reply:
Done.